akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -217,22 +219,7 @@ object GraphStageLogic {
         case pair => callback.invoke(pair)
       }
 
-      materializer.supervisor match {
-        case ref: LocalActorRef =>
-          ref.underlying.addFunctionRef(f, name)
-        case ref: RepointableActorRef =>
-          if (ref.isStarted)
-            ref.underlying.asInstanceOf[ActorCell].addFunctionRef(f, name)
-          else {
-            // this may happen if materialized immediately before Materializer has been fully initialized,
-            // should be rare
-            implicit val timeout = ref.system.settings.CreationTimeout
-            val reply = (materializer.supervisor ? StreamSupervisor.AddFunctionRef(f, name)).mapTo[FunctionRef]
-            Await.result(reply, timeout.duration)
-          }
-        case unknown =>
-          throw new IllegalStateException(s"Stream supervisor must be a local actor, was [${unknown.getClass.getName}]")
-      }
```

Comment:
No longer needed, since materializer is now always a child of a "normal" actor, it will never be anything but a `LocalActorRef`